Fix #522 - Add --start-in-tray CLI flag to let app start in background

tribut opened the pull request:
This fixes #522 by introducing `--start-in-tray` as new CLI flag.

ronjouch commented:
@tribut ah yes, Commander sucks for that indeed. But I'm tired of more<sup>more</sup> flags, and I certainly do not want to harm backwards compat. So let's just work around; this should work:

```js
#!/usr/bin/env node

var program = require('commander');

const sanitizedArgs = [];
process.argv.forEach(arg => {
  if (sanitizedArgs.length > 0) {
    const previousArg = sanitizedArgs[sanitizedArgs.length - 1];

    // Work around commander.js not supporting default argument for options
    if (previousArg === '--tray' && !['true', 'false', 'start-in-tray'].includes(arg)) {
      sanitizedArgs.push('true');
    }
  }
  sanitizedArgs.push(arg);
});

program
  .option('--tray [true|false|start-in-tray]', 'Use tray icon', 'false')
  .parse(sanitizedArgs);

console.log('tray:', program.tray);
```

Result:

```bash
~ ./test.js --tray true 'http://google.com/' # explicit true
tray: true

~ ./test.js --tray false 'http://google.com/' # explicit false
tray: false

~ ./test.js --tray start-in-tray 'http://google.com/' # explicit start-in-tray
tray: start-in-tray

~ ./test.js --tray 'http://google.com/' # implicit true (backward compat)
tray: true

~ ./test.js 'http://google.com/' # implicit false (backward compat)
tray: false
```

Also, please rebase / resolve conflicts. Then I'll merge.

tribut commented:
I've now added your workaround and rebased. I've also added a function to make sure `--tray false` works (`"true" == true`, so that worked without explicit conversion) - not sure if that is needed, but given you had true and false in your workaround I figured.

Note that codeclimate and travis seem to disagree whether the parenthesis around `arg` in https://github.com/jiahaog/nativefier/blob/cd07d9e0145e3847ac80ee706587b58af6a60b6f/src/cli.js#L51 are needed. I've sided with travis and added them in ¯\_(ツ)_/¯

ronjouch commented on src/cli.js:
```diff
@@ -13,6 +13,17 @@ function collect(val, memo) {
   return memo;
 }
 
+function parseMaybeBoolString(val) {
```
👍

ronjouch commented on src/cli.js:
```diff
@@ -36,6 +47,22 @@ function checkInternet() {
 }
 
 if (require.main === module) {
+  const sanitizedArgs = [];
```
@tribut but you need to use the value at line 237: `.parse(process.argv);` should become `.parse(sanitizedArgs);`

That suggests you didn't test these last changes. Can you do a quick check and validate that all three options work as expected? Thanks 🙂.

ronjouch commented on src/cli.js:
```diff
@@ -36,6 +47,22 @@ function checkInternet() {
 
 if (require.main === module) {
+  const sanitizedArgs = [];
+  process.argv.forEach((arg) => {
```
Yeah you're right to do what ESLint asks for.
The weird thing is that CodeClimate report this as an ESLint error.
Dunno what's going on. Disabling the eslint CodeClimate plugin, we have it in the regular `ci` npm task.

ronjouch replied:
Okay, I pushed a fix in your branch, CodeClimate and Travis are both happy now. Waiting for your last fix and test.

tribut commented:
Thanks for fixing the eslint issue. Not sure how my changes went missing, but probably during the several round of style changes. Rebased and tested again now and [no argument]/true/false/start-in-tray do all work for me.

Sparsa commented:
Just trying this in-tray feature, can you please explain how to deal with this if I want to make an application start on tray? I made an application using nativefier --tray start-in-tray option. Now, when I start the application, it comes to tray but also starts maximized.
